src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1944,14 +1944,32 @@
     <Field Name="LengthOrCount" Type="BoundExpression?" SkipInVisitor="true" Null="allow"/>
     <!-- Collection element placeholder, used in IteratorBody. -->
     <Field Name="ElementPlaceholder" Type="BoundValuePlaceholder?" SkipInVisitor="true" Null="allow"/>
-    <!-- Statement executed for each collection element. -->
-    <Field Name="IteratorBody" Type="BoundStatement?" SkipInVisitor="true" Null="allow"/>
+    <!-- Code executed for each collection element. -->
+    <Field Name="IteratorBody" Type="BoundNode?" SkipInVisitor="true" Null="allow"/>
   </Node>
 
   <!-- A key:value pair element within a collection expression. -->
   <Node Name="BoundKeyValuePairElement" Base="BoundNode">
```

Comment:
nit: I initially found it surprising to have an assignment and placeholders in this node, but it may be a question of naming. Maybe this should be `BoundIndexerAssignmentFromKeyValuePairElement`?

Reply:
I'm not sure I can fully flesh the idea just yet, but I wonder if there's more factoring to be done here. 
Instead of having two different bound nodes for "indexer assignment", could we abstract a bound node that is a source of key and value parts?
Then we can just have a single BoundIndexerAssignment node, but it can have a source for those parts that's either: 1) pre-computed (ie. `k: p` scenario) or 2) computed (ie. getting `.Key` and `.Value` from the expression).

Reply:
From offline discussion, we'd also need to account for the spread scenario.
We settled instead on simply renaming "BoundIndexerAssignmentFromExpressionElement" to "BoundKeyValuePairExpression" or "BoundKeyValuePairExpressionElement".